src/main/java/org/ecommerce/enums/CategoryType.java

```diff
@@ -1,5 +1,5 @@
 package org.ecommerce.enums;
 
-public enum Category {
+public enum CategoryType {
```

Comment:
When do we use `CategoryType`? Can a `category` have multiple `children` and, furthermore, `grand children`?

Reply:
Yeah, we actually didn't conclude anything with Mario about that, but at the beginning we were thinking in the implementation of subcategories. A category with multiple ```sub-categories``` or ```childrens```, that was the originally purpose of that field.

Reply:
Hehehe I see.
A product usually has multiple categories. So a String array/list might be okay. However I think you can keep your original structure as we won't dig much into the category implementation/usage anyway.